core/src/main/java/io/questdb/cairo/vm/AbstractMemoryCR.java

```diff
@@ -27,19 +27,15 @@
 import io.questdb.cairo.CairoException;
 import io.questdb.cairo.vm.api.MemoryCR;
 import io.questdb.std.*;
-import io.questdb.std.str.DirectCharSequence;
-import io.questdb.std.str.DirectString;
-import io.questdb.std.str.DirectUtf8String;
-import io.questdb.std.str.Utf8Sequence;
-import io.questdb.std.str.Utf8SplitString;
+import io.questdb.std.str.*;
 import org.jetbrains.annotations.NotNull;
 
 // contiguous readable
 public abstract class AbstractMemoryCR implements MemoryCR, Mutable {
 
     private final MemoryCR.ByteSequenceView bsview = new MemoryCR.ByteSequenceView();
-    private final DirectString csviewA = new DirectString();
-    private final DirectString csviewB = new DirectString();
+    private final StableDirectString csviewA = new StableDirectString();
```

Comment:
Stable string class should be used only in the memory used by `TableReader`. That's because we have `MemoryFCRImpl` and some other non-mmapped memory classes extending `AbstractMemoryCR`.

Reply:
@puzpuzpuz can you have another look? I believe what we have now should work. but the elegance of the original idea (=everything encapsulating inside a 'sink') is gone:(. the 'support directs str' is leaking again :(